CMakeLists.txt

```diff
@@ -59,14 +66,16 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     if(RAPIDJSON_ENABLE_INSTRUMENTATION_OPT AND NOT CMAKE_CROSSCOMPILING)
         if(CMAKE_SYSTEM_PROCESSOR STREQUAL "powerpc" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "ppc64" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "ppc64le")
           set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mcpu=native")
-        else()
+        elseif(NOT ANDROID)
```

Comment:
I added this to handle https://github.com/Tencent/rapidjson/commit/ffe3a3daa80d49ada0a786225d1935048e4e5ada. Note that the `-W*` flags are already set below for all.